scene/gui/code_edit.cpp

```diff
@@ -1488,6 +1494,22 @@ bool CodeEdit::can_fold_line(int p_line) const {
 		return false;
 	}
 
+	/* Check for code region. */
+	if (is_line_code_region_closing_tag(p_line)) {
+		return false;
+	}
+	if (is_line_code_region_opening_tag(p_line)) {
+		/* Check if there is a valid end region tag */
+		for (int next_line = p_line + 1; next_line < get_line_count(); next_line++) {
+			if (is_line_code_region_closing_tag(next_line)) {
+				return true;
+			} else if (is_line_code_region_opening_tag(next_line)) {
+				WARN_PRINT_ONCE(vformat("Nested code regions are not supported. (Lines %s and %s)", p_line, next_line));
+				return false;
+			}
+		}
```

Comment:
Can we early return here?

Reply:
I think so because if we allow to fold a nested region start line, it will fold until the next endregion even there is another region start between these two lines, it's sure that we do not want that.

Alternative solution is to allow nested region and keep track of a count of tags for detecting corresponding region tag.